packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -508,7 +508,7 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 		const newCommits = targetCommits.concat(sourceCommits);
 		if (this.isTransacting()) {
 			const src = targetCommits[0].parent?.revision;
-			const dst = targetCommits.at(-1)?.revision;
+			const dst = targetCommits[targetCommits.length - 1]?.revision;
```

Comment:
this code seems to explicitly handle this index being out of bounds (due to empty array).

given that the previous line would throw in that case, I think you should do:
```suggestion
			const dst = targetCommits[targetCommits.length - 1].revision;
```